packages/sources/dxfeed/src/endpoint/price.ts

```diff
@@ -56,15 +58,19 @@ export const execute: ExecuteWithConfig<Config> = async (request, _, config) =>
 
   // NOTE: may need to force entries quoteEventSymbols to not use batching
 
-  const quotePath = ['Quote', symbol, 'bidPrice']
-  const tradePath = ['Trade', symbol, 'price']
   response.data.result = Requester.validateResultNumber(
     response.data,
-    events === 'Quote' ? quotePath : tradePath,
+    getResultPath(events, symbol),
   )
   return Requester.success(jobRunID, response, config.verbose, batchablePropertyPath)
 }
 
+const getResultPath = (events: Events, symbol: string, isArray = false): Array<string | number> => {
```

Comment:
What if we turned this into the way that other adapters use resultPath from `builder.ts` by exporting `endpointResultPaths`?
It accepts a function to handle dynamic resultPaths like this.

https://github.com/smartcontractkit/external-adapters-js/blob/develop/packages/sources/coingecko/src/endpoint/crypto.ts#L27
shows how to export it by endpoint, so using a similar pattern of a factory function this one would look like:
```
export const endpointResultPaths: {
  [endpoint: string]: ReturnType<typeof getResultPath>
} = {
  price: getResultPath(),
  crypto: getResultPath(),
  stock: getResultPath(),
  forex: getResultPath(),
  commodities: getResultPath(),
}
```

`getResultPath` would turn into something like:
```
const getResultPath = () => (request: AdapterRequest) => {
  const validator = new Validator(request, inputParameters)
  if (validator.error) throw validator.error
  const base = validator.validated.data.base
 const events = quoteEventSymbols[symbol] ? 'Quote' : 'Trade' (could abstract this into its own function)
  if (Array.isArray(base)) return
  return [events, symbol, events === 'Quote' ? 'bidPrice' : 'price']
}
```

Reply:
The benefit of this is that the core framework has access to the `resultPath`, which will be useful once we start caching full data responses.

Reply:
This was already in there - but if the base is batched and included `'USO/USD:AFX'` it would never be picked up by `quoteEventSymbols` because it will key in with all of the comma delimited bases.

That's one reason to not pass through `events` into handleBatchedRequest and instead use this new `getResultPath` repeatedly.

Reply:
I've addressed the other comments, but didn't make these changes, I'm not fully understanding the desired outcome of this